test/remote-models.test.js

```diff
@@ -11,25 +11,97 @@ var TaskEmitter = require('strong-task-emitter');
 
 describe('Remote model tests', function() {
   var ctx = this;
-
+  var memoryDs = helper.createMemoryDataSource();
   beforeEach(function(done) {
+    memoryDs.connector.cache = {}; // empty cache of memory connector
     ctx.serverApp = helper.createRestAppAndListen();
     ctx.ServerModel = helper.createModel({
       parent: 'TestModel',
       app: ctx.serverApp,
-      datasource: helper.createMemoryDataSource(),
-      properties: helper.userProperties
+      datasource: memoryDs,
+      properties: helper.getUserProperties(),
+      options: {
+        relations: {
+          children: {
+            type: 'hasMany',
+            model: 'ChildModel',
+            foreignKey: 'parentId'
+          }
+        }
+      }
+    });
+    ctx.ServerModel2 = helper.createModel({
+      parent: 'TestModel2',
+      app: ctx.serverApp,
+      datasource: memoryDs,
+      properties: helper.getUserProperties(),
+      options: {
+        relations: {
+          child: {
+            type: 'hasOne',
+            model: 'ChildModel',
+            foreignKey: 'parentId'
+          }
+        }
+      }
+    });
+    ctx.ServerRelatedModel = helper.createModel({
+      parent: 'ChildModel',
+      app: ctx.serverApp,
+      datasource: memoryDs,
+      properties: {
+        note: { type: 'text' },
+        parentId: { type: 'number' }
+      }
     });
     ctx.serverApp.locals.handler.on('listening', function() { done(); });
   });
 
   beforeEach(function setupRemoteClient(done) {
     ctx.remoteApp = helper.createRestAppAndListen();
+    ctx.RemoteRelatedModel = helper.createModel({
+      parent: 'ChildModel',
+      app: ctx.remoteApp,
+      datasource: memoryDs,
```

Comment:
FWIW, since this model is attached to the client ("remote") app, it should use a remote datasource too. I can fix this myself after #82 is landed.